a3p-integration/proposals/z:acceptance/test-lib/vaults.js

```diff
@@ -0,0 +1,233 @@
+  console.log('availableDebtForMint: ', availableDebtForMint);
+
+  return {
+    availableDebtForMint,
+    debtLimit: debtLimit.value,
+    totalDebt: totalDebt.value,
+  };
+};
+
+/**
+ *
+ * @returns {Promise<bigint>}
+ */
+export const getMinInitialDebt = async () => {
+  const governancePath = `published.vaultFactory.governance`;
+  const governance = await getContractInfo(governancePath, {
+    agoric,
+    prefix: '',
+  });
+
+  const minInitialDebt = governance.current.MinInitialDebt.value.value;
+  console.log('minInitialDebt: ', minInitialDebt);
+
+  return minInitialDebt / 1_000_000n;
+};
+
+/**
+ *
+ * @param {bigint} toMintValue
+ * @param {string} vaultManager
+ * @returns {Promise<{ mintFee: import('@agoric/ertp/src/types.js').NatAmount, adjustedToMintAmount: import('@agoric/ertp/src/types.js').NatAmount }>}
+ */
+export const calculateMintFee = async (toMintValue, vaultManager) => {
+  const brands = await getAgoricNamesBrands();
+
+  const governancePath = `published.vaultFactory.managers.${vaultManager}.governance`;
+  const governance = await getContractInfo(governancePath, {
+    agoric,
+    prefix: '',
+  });
+
+  const mintFee = governance.current.MintFee;
+  const { numerator, denominator } = mintFee.value;
+  const mintFeeRatio = harden({
+    numerator: AmountMath.make(brands.IST, numerator.value),
+    denominator: AmountMath.make(brands.IST, denominator.value),
+  });
+
+  const toMintAmount = AmountMath.make(brands.IST, toMintValue * 1_000_000n);
+  const expectedMintFee = ceilMultiplyBy(toMintAmount, mintFeeRatio);
+  const adjustedToMintAmount = AmountMath.add(toMintAmount, expectedMintFee);
+
+  console.log('mintFee: ', mintFee);
+  console.log('adjustedToMintAmount: ', adjustedToMintAmount);
+
+  return { mintFee, adjustedToMintAmount };
+};
+
+const voteForNewParams = (accounts, position) => {
+  console.log('ACTIONS voting for position', position, 'using', accounts);
+  return Promise.all(
+    accounts.map(account =>
+      agops.ec('vote', '--forPosition', position, '--send-from', account),
+    ),
+  );
+};
+
+const paramChangeOfferGeneration = async (
+  previousOfferId,
+  voteDur,
+  debtLimit,
+) => {
+  const ISTunit = 1_000_000n; // aka displayInfo: { decimalPlaces: 6 }
+
+  const brand = await getAgoricNamesBrands();
+  assert(brand.IST);
+  assert(brand.ATOM);
+
+  const instance = await getAgoricNamesInstances();
+  assert(instance.VaultFactory);
+
+  const voteDurSec = BigInt(voteDur);
+  const debtLimitValue = BigInt(debtLimit) * ISTunit;
+  const toSec = ms => BigInt(Math.round(ms / 1000));
+
+  const id = `propose-${Date.now()}`;
+  const deadline = toSec(Date.now()) + voteDurSec;
+
+  const body = {
+    method: 'executeOffer',
+    offer: {
+      id,
+      invitationSpec: {
+        invitationMakerName: 'VoteOnParamChange',
+        previousOffer: previousOfferId,
+        source: 'continuing',
+      },
+      offerArgs: {
+        deadline,
```

Comment:
This function was indeed extracted from an existing test helper file [vaults.mts](https://github.com/Agoric/agoric-sdk/blob/master/a3p-integration/proposals/z%3Aacceptance/lib/vaults.mts#L65-L146), and I agree with your observation that this `Date.now()` behavior appears consistently throughout the `a3p-integration` tests. 
It would make sense to open an issue to refactor these tests, provided there is agreement to move toward using `chainTimer` and `timeMath` for consistency and accuracy in deadline calculations.

As for whether the `chainTimer` itself is on wall-clock time, I’m not sure, but I will dig deeper on the subject

Reply:
cc: @anilhelvaci